src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.ts

```diff
@@ -480,7 +454,11 @@ export class ProgrammingExerciseDetailComponent implements OnInit, OnDestroy {
      * Returns undefined if the last solution submission was not successful or no report exists yet
      */
     private setLatestCoveredLineRatio() {
-        const latestSolutionSubmissionSuccessful = this.getLatestResult(this.programmingExercise?.solutionParticipation?.submissions)?.successful;
+        if (!this.programmingExercise?.solutionParticipation) {
+            return;
+        }
+
+        const latestSolutionSubmissionSuccessful = this.programmingExerciseService.getLatestResult(this.programmingExercise?.solutionParticipation)?.successful;
```

Comment:
The check above ensures that this is not undefined.

```suggestion
        const latestSolutionSubmissionSuccessful = this.programmingExerciseService.getLatestResult(this.programmingExercise.solutionParticipation)?.successful;
```